api/prometheus/v1/api.go

```diff
@@ -139,6 +165,38 @@ type SnapshotResult struct {
 	Name string `json:"name"`
 }
 
+// RulesResult contains the result from querying the rules endpoint.
+type RulesResult struct {
+	Groups []RuleGroup `json:"groups"`
+}
+
+type RuleGroup struct {
+	Name     string  `json:"name"`
+	File     string  `json:"file"`
+	Rules    []Rule  `json:"rules"`
+	Interval float64 `json:"interval"`
+}
+
+type Rule struct {
```

Comment:
The name `Rule` is misleading, as recording rules do not expose e.g. `Alerts`.

Reply:
Again, could we import structs from prometheus/prometheus instead?

Reply:
I think we should just create two new types here: `RecordingRule` and `AlertingRule`. Is there any reason why order needs to be preserved by putting both types of rules in the same array?

If we create these two new types, I think we'll need some changes to the imported structs, for example `recordingRule` and `alertingRule` are not exported:
https://github.com/prometheus/prometheus/blob/master/web/api/v1/api.go#L730
https://github.com/prometheus/prometheus/blob/master/web/api/v1/api.go#L743